OpenRA.Game/MiniYaml.cs

```diff
@@ -215,64 +243,84 @@ static List<MiniYamlNode> FromLines(IEnumerable<string> lines, string filename,
 					// Leading and trailing whitespace is always trimmed from keys.
 					// Leading and trailing whitespace is trimmed from values unless they
 					// are marked with leading or trailing backslashes
-					var keyLength = line.Length - keyStart;
+					var keyLength = lineLength - keyStart;
 					var valueStart = -1;
 					var valueLength = 0;
 					var commentStart = -1;
-					for (var i = 0; i < line.Length; i++)
+					var hasEscapedHashChar = false;
+					var prevChar = '\0';
+					for (var i = keyStart; i < lineLength; i++)
 					{
-						if (valueStart < 0 && line[i] == ':')
+						var currChar = line[i];
+						if (commentStart < 0)
```

Comment:
We can drop this check now that we `break` after setting a `commentStart` value.